frontend/javascripts/oxalis/model/sagas/volumetracing_saga.js

```diff
@@ -133,14 +137,24 @@ export function* editVolumeLayerAsync(): Generator<any, any, any> {
       }
       if (activeTool === VolumeToolEnum.BRUSH) {
         const currentViewportBounding = yield* call(getBoundingsFromPosition, activeViewport);
+        const rectangleIterator = lastPosition
+          ? currentLayer.getRectangleVoxelIterator(
+              lastPosition,
+              addToLayerAction.position,
+              currentViewportBounding,
+            )
+          : null;
+        if (rectangleIterator) {
+          yield* call(labelWithIterator, rectangleIterator, contourTracingMode);
```

Comment:
This means that the rectangle is always drawn between two points, right? In the original issue description, I put the idea that the rectangle drawing might be only necessary if the distance between the two brush positions is greater than a certain threshold. This could save the entire rectangle drawing if the user moves the mouse slowly (--> better performance).
Maybe you could try this out? E.g., threshold = radius or threshold = radius/10. There will definitely be a trade-off between how the brush stroke will look and how much time is saved. Maybe it doesn't save much performance, but might be worth to measure.

Reply:
The decision whether to paint the rectangle or not is made before drawing the rectangle, but in the drawing method. I put it there so that I could reuse some of the necessary calculations when drawing the rectangle.